ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/CommandQueue.java

```diff
@@ -48,29 +52,32 @@ QueuedCommandStatus enqueueCommand(
   );
   
   /**
-   * Polls the Queue for any commands that have been enqueued since the last
-   * invocation to this method.
+   * Polls the Queue for any command records that have been enqueued since the last
+   * invocation to this method. Returns the CommandStatusFuture associated with a record
+   * if it exists
    *
    * <p>The method blocks until either there is data to return or the
    * supplied {@code timeout} expires.
    *
    * <p>If between invocations to this method, {@link #getRestoreCommands()} is
    * invoked, this command will begin where the results of that call ended.
    *
-   * @param timeout the max time to wait for new commands.
-   * @return a list of commands that have been enqueued since the last call
+   * @param timeout the max time to wait for new records.
+   * @return a list of records that have been enqueued since the last call
    * @apiNote this method may block
    */
-  List<QueuedCommand> getNewCommands(Duration timeout);
```

Comment:
I don't agree with the decision to pull QueuedCommand out of this interface and replace it with Pair<ConsumerRecord, CommandStatusFuture>> since it breaks this abstraction unnecessarily. The idea behind this interface was to provide an api to a message queue where a caller could enqueue/dequeue, and also register to be notified when a given message is consumed/processed, or when a given offset is consumed/processed. It's not important that the backing store is Kafka, and that shouldn't really leak into the interface (I know the txnl producer is in there, but that can be fixed). It's also not that important that the messages have key/value types CommandID/Command (I know, bad naming 😁 ). Therefore I think it would be better to organize this in the following way:

Leave this interface mostly the same, except change references to Command and CommandID to byte[]

Change QueuedCommand's members to be:
```
class QueuedCommand {
    byte[] commandId;
    byte[] command;
    ....
}
```

That said, it is useful to have an explicit type for Command to improve safety (so for example, a caller can't accidentally mix them up, or CommandStore can't accidentally use the Command as the key to the status map). So, an additional improvement we could make would be to leave the `command` parameters as type Command (I'd still use byte[] for the command id and just serialize/deserialize that from the caller). Then, in QueuedCommand, change `getCommand` to `getAndDeserializeCommand(Deserializer<Command> deserializer)` so that CommandRunner is the entity that actually does the deserialization.

Reply:
I used your suggestion as a guideline and have updated the PR. (No changes to CommandQueue)